src/PowerShellEditorServices/Language/FindDotSourcedVisitor.cs

```diff
@@ -13,14 +17,17 @@ namespace Microsoft.PowerShell.EditorServices
     /// </summary>
     internal class FindDotSourcedVisitor : AstVisitor
     {
-       /// <summary>
-       /// A hash set of the dot sourced files (because we don't want duplicates)
-       /// </summary>
+        private readonly string _scriptDirectory;
+
+        /// <summary>
+        /// A hash set of the dot sourced files (because we don't want duplicates)
+        /// </summary>
         public HashSet<string> DotSourcedFiles { get; private set; }
 
-        public FindDotSourcedVisitor()
+        public FindDotSourcedVisitor(string scriptPath)
```

Comment:
Might be worth a comment that the only thing `scriptPath` does is provide the script dir for substitution of `$PSScriptRoot`.  In fact, maybe the caller should just provide that info in which case I'd rename this to `scriptRoot` or `psScriptRoot`.

Reply:
On the one hand, the caller could amortise the `$PSScriptRoot` calculation. On the other hand, the information might not make much sense being stored by the caller from a "cohesion" perspective.

Reply:
Yeah, I could go either way but a comment explaining what the value is used for would have saved us both from rooting around in the code to see what it does.  :-)

Reply:
In another iteration that was never committed I also handled `$PSCommandPath` and sending the file path allowed me to calculate both that and `$PSScriptRoot`.

I'll add the comment!

Reply:
Looks like ScriptFiles are made on an individual basis, so not much saving there:
https://github.com/PowerShell/PowerShellEditorServices/blob/11a3107cbe4753ffaca2bba89acf020160e432eb/src/PowerShellEditorServices/Workspace/Workspace.cs#L86-L128

With that said, I do think it makes sense to allocate the scriptroot string once per script file and pass it in, rather than every function call.

Reply:
Heheh, I was going to ask about `PSCommandPath` but didn't think it would be used in the dot source scenarios you're addressing.

Reply:
And I think calling it `psScriptRoot` makes it clear enough that only a small comment is needed

Reply:
Done